appformer-ala-wildfly-provider/src/main/java/org/kie/appformer/ala/wildfly/util/JarUtils.java

```
@@ -0,0 +1,154 @@
+            safeClose( jarFile );
+            throw e;
+        }
+
+        try {
+            // add the new entry.
+            JarEntry newJarEntry = new JarEntry( entry );
+            tmpJarOutputStream.putNextEntry( newJarEntry );
+            tmpJarOutputStream.write( content.getBytes( ) );
+
+            //Copy the original entries into the result skipping the new one.
+            Enumeration< JarEntry > entries = jarFile.entries( );
+            JarEntry jarEntry;
+            while ( entries.hasMoreElements( ) ) {
+                jarEntry = entries.nextElement( );
+                if ( !jarEntry.getName( ).equals( entry ) ) {
+                    InputStream entryInputStream = jarFile.getInputStream( jarEntry );
+                    tmpJarOutputStream.putNextEntry( jarEntry );
+                    byte[] buffer = new byte[ 1024 ];
+                    int bytesRead;
+                    while ( ( bytesRead = entryInputStream.read( buffer ) ) != -1 ) {
+                        tmpJarOutputStream.write( buffer, 0, bytesRead );
+                    }
+                    entryInputStream.close( );
+                }
+            }
+        } catch ( IOException e ) {
+            logger.error( "It was not possible to create the tmp target file content for file:" + tmpJarFilePath.toUri( ), e );
+            throw e;
+        } finally {
+            safeClose( jarFile );
+            safeClose( tmpJarOutputStream );
+        }
+
+        try {
+            //swap the files.
+            Files.copy( tmpJarFilePath, jarFilePath, StandardCopyOption.REPLACE_EXISTING );
+        } catch ( Exception e ) {
+            logger.error( "It was not possible to replace file content for file: " + jarFilePath.toUri( ) );
+            throw e;
+        } finally {
+            try {
+                Files.delete( tmpJarFilePath );
+            } catch ( Exception e ) {
+                logger.warn( "An error was produced during tmp file removal: " + tmpJarFilePath );
```

This will effectively swallow the original exception, would you consider adding it as another parameter to logger.warn() or log it otherwise (maybe on a different log level if there is concern about the log being cluttered with stacktraces)?